packages/editor/src/plugins/image/components/licence-dropdown.tsx

```diff
@@ -0,0 +1,118 @@
+import { faQuestionCircle } from '@fortawesome/free-regular-svg-icons'
+import { faCaretDown, faCaretUp } from '@fortawesome/free-solid-svg-icons'
+import * as Select from '@radix-ui/react-select'
+import React, { useState, useEffect, useRef } from 'react'
+
+import { FaIcon } from '@/components/fa-icon'
+import { useEditorStrings } from '@/contexts/logged-in-data-context'
+import { cn } from '@/helper/cn'
+
+interface LicenseDropdownProps {
+  onLicenseChange?: (license: string) => void
+  src?: string
+  currentLicence?: string
+}
+
+export const LicenseDropdown = ({
+  onLicenseChange,
+  src,
+  currentLicence,
+}: LicenseDropdownProps) => {
+  const editorStrings = useEditorStrings()
+  const imageStrings = editorStrings.plugins.image
+  const { licences, licenceHelpText } = imageStrings
+
+  const licenceNames = Object.values(licences)
+
+  const [selectedLicense, setSelectedLicense] = useState(
+    currentLicence ?? 'CC4'
+  )
+  const [isConfirmed, setIsConfirmed] = useState(false)
+  const [isOpen, setIsOpen] = useState(false)
+  const [triggerWidth, setTriggerWidth] = useState(0)
+
+  const triggerRef = useRef<HTMLLabelElement>(null)
+
+  const isPixabayImage = src?.includes('pixabay')
+  useEffect(() => {
+    if (isPixabayImage) {
+      const pixabayLicence =
+        licenceNames.find((name) => name.toLowerCase().includes('pixabay')) ||
+        ''
+      setSelectedLicense(pixabayLicence)
+      onLicenseChange?.(pixabayLicence)
+      setIsConfirmed(true)
+    }
+  }, [isPixabayImage, onLicenseChange, licenceNames])
+
+  useEffect(() => {
+    setTriggerWidth(triggerRef.current ? triggerRef.current.offsetWidth / 3 : 0)
+  }, [])
+
+  const handleLicenseChange = (newLicence: string) => {
+    setSelectedLicense(newLicence)
+    onLicenseChange?.(newLicence)
+    setIsConfirmed(true)
+  }
+
+  return (
+    <label
+      ref={triggerRef}
+      className="mx-auto mb-0 mt-5 flex items-center text-almost-black"
+    >
+      <span className="serlo-tooltip-trigger w-1/4">
+        {imageStrings.licence}
+        <FaIcon className="ml-2" icon={faQuestionCircle} />
+        <EditorTooltip text={licenceHelpText} />
+      </span>
+      <span className="flex w-3/4 flex-row border-2 border-transparent">
+        <Select.Root
+          onValueChange={handleLicenseChange}
+          defaultValue={selectedLicense}
+          onOpenChange={setIsOpen}
+          value={selectedLicense}
+        >
+          <Select.Trigger
+            className={cn(
+              'serlo-input-font-reset rounded-md border-2 border-editor-primary-100 bg-editor-primary-100 px-2 text-left focus:border-2 focus:border-editor-primary',
+              'flex flex-row justify-between',
+              isOpen &&
+                'rounded-b-none border-editor-primary-300 border-b-transparent'
+            )}
+            style={{ width: `${triggerWidth}px` }}
+          >
+            <Select.Value placeholder="Select an option" />
```

Comment:
Not sure if this placeholder prop will ever be visible. If it will, could you use an i18n string? If it won't, maybe remove it?

Reply:
Removed